Index: backend/cgobj.c
===================================================================
--- backend/cgobj.c	(版本 804)
+++ backend/cgobj.c	(工作副本)
@@ -1190,7 +1190,7 @@
     int i;
     int x2;
 
-    printf("obj_wkext(%s)\n", s1->Sident);
+    //dbg_printf("obj_wkext(%s)\n", s1->Sident);
     if (s2)
         x2 = s2->Sxtrnnum;
     else
Index: toobj.c
===================================================================
--- toobj.c	(版本 804)
+++ toobj.c	(工作副本)
@@ -151,7 +151,7 @@
                  * they resolve to 0 if not pulled in by something else.
                  * Don't pull in a module just because it was imported.
                  */
-#if !OMFOBJ // Optlink crashes with weak symbols at EIP 41AFE7, 402000
+#if 1 || !OMFOBJ // Optlink crashes with weak symbols at EIP 41AFE7, 402000
                 s->Sflags |= SFLweak;
 #endif
                 dtxoff(&dt, s, 0, TYnptr);
Index: expression.c
===================================================================
--- expression.c	(版本 804)
+++ expression.c	(工作副本)
@@ -44,7 +44,7 @@
 #include "hdrgen.h"
 #include "parse.h"
 
-
+size_t plugin_import(char*, char**, void**);
 Expression *createTypeInfoArray(Scope *sc, Expression *args[], int dim);
 Expression *expandVar(int result, VarDeclaration *v);
 
@@ -5659,7 +5659,19 @@
     {   error("need -Jpath switch to import text file %s", name);
         goto Lerror;
     }
-
+    
+    size_t plugin_len;
+    char* plugin_error;
+    void* plugin_buffer ;
+    plugin_len	= plugin_import(name, &plugin_error, &plugin_buffer);
+    if( plugin_error ) {
+        error("%s", plugin_error);
+        goto Lerror; 
+    } else if( plugin_len!= 0 ) {
+        se = new StringExp(loc, plugin_buffer, plugin_len);
+        goto Lret;
+    }
+    
     /* Be wary of CWE-22: Improper Limitation of a Pathname to a Restricted Directory
      * ('Path Traversal') attacks.
      * http://cwe.mitre.org/data/definitions/22.html
Index: module.c
===================================================================
--- module.c	(版本 804)
+++ module.c	(工作副本)
@@ -239,7 +239,9 @@
     {   error("Source file and 'header' file have same name '%s'", srcfile->name->str);
         fatal();
     }
-
+	if( !global.params.hdrdir  && !global.params.hdrname ) {
+		hdrfilename = FileName::forceExt(srcfile->toChars() , global.hdr_ext);
+	}
     hdrfile = new File(hdrfilename);
 }
 
Index: win32.mak
===================================================================
--- win32.mak	(版本 804)
+++ win32.mak	(工作副本)
@@ -4,12 +4,12 @@
 # All Rights Reserved
 # Build dmd with Digital Mars C++ compiler
 
-D=
+D=Z:
 DMDSVN=\svnproj\dmd\trunk\src
 #DMDSVN=\svnproj\dmd\branches\dmd-1.x\src
 SCROOT=$D\dm
 INCLUDE=$(SCROOT)\include
-CC=\dm\bin\dmc
+CC=Z:\dm\bin\dmc
 LIBNT=$(SCROOT)\lib
 SNN=$(SCROOT)\lib\snn
 DIR=\dmd2
@@ -47,9 +47,9 @@
 ################ RELEASES #########################
 
 release:
-	$(MAKE) clean
 	$(MAKE) dmd
-	$(MAKE) clean
+	copy dmd.exe Z:\dmd\bin\dmd2.exe
+	del *.exe *.map
 
 ################ NT COMMAND LINE RELEASE #########################
 
@@ -81,7 +81,7 @@
 	builtin.obj clone.obj libomf.obj arrayop.obj irstate.obj \
 	glue.obj msc.obj ph.obj tk.obj s2ir.obj todt.obj e2ir.obj tocsym.obj \
 	util.obj bit.obj eh.obj toobj.obj toctype.obj tocvdebug.obj toir.obj \
-	json.obj unittests.obj imphint.obj argtypes.obj
+	json.obj unittests.obj imphint.obj argtypes.obj plugin.obj
 
 # from C/C++ compiler optimizer and back end
 
@@ -115,7 +115,7 @@
 	macro.h macro.c hdrgen.h hdrgen.c arraytypes.h \
 	delegatize.c toir.h toir.c interpret.c traits.c builtin.c \
 	clone.c lib.h libomf.c libelf.c libmach.c arrayop.c \
-	aliasthis.h aliasthis.c json.h json.c unittests.c imphint.c argtypes.c
+	aliasthis.h aliasthis.c json.h json.c unittests.c imphint.c argtypes.c  plugin.c
 
 # From C++ compiler