diff --git a/src/backend/cgobj.c b/src/backend/cgobj.c
index 2590904..1d7e180 100644
--- a/src/backend/cgobj.c
+++ b/src/backend/cgobj.c
@@ -1178,7 +1178,7 @@ void obj_wkext(Symbol *s1,Symbol *s2)
     int i;
     int x2;
 
-    printf("obj_wkext(%s)\n", s1->Sident);
+    dbg_printf("obj_wkext(%s)\n", s1->Sident);
     if (s2)
         x2 = s2->Sxtrnnum;
     else
diff --git a/src/expression.c b/src/expression.c
index e0be13c..84096d5 100644
--- a/src/expression.c
+++ b/src/expression.c
@@ -46,6 +46,7 @@ extern "C" char * __cdecl __locale_decpoint;
 #include "parse.h"
 #include "doc.h"
 
+size_t plugin_import(char*, char**, void**);
 
 Expression *createTypeInfoArray(Scope *sc, Expression *args[], unsigned dim);
 Expression *expandVar(int result, VarDeclaration *v);
@@ -6142,6 +6143,18 @@ Expression *FileExp::semantic(Scope *sc)
         goto Lerror;
     }
 
+    size_t plugin_len;
+    char* plugin_error;
+    void* plugin_buffer ;
+    plugin_len	= plugin_import(name, &plugin_error, &plugin_buffer);
+    if( plugin_error ) {
+        error("%s", plugin_error);
+        goto Lerror; 
+    } else if( plugin_len!= 0 ) {
+        se = new StringExp(loc, plugin_buffer, plugin_len);
+        return se->semantic(sc);
+    }
+
     /* Be wary of CWE-22: Improper Limitation of a Pathname to a Restricted Directory
      * ('Path Traversal') attacks.
      * http://cwe.mitre.org/data/definitions/22.html
diff --git a/src/module.c b/src/module.c
index 3074e15..c048970 100644
--- a/src/module.c
+++ b/src/module.c
@@ -209,6 +209,10 @@ void Module::setDocfile()
     {   error("Source file and documentation file have same name '%s'", srcfile->name->str);
         fatal();
     }
+    
+	if( !global.params.hdrdir  && !global.params.hdrname ) {
+		docfilename = FileName::forceExt(srcfile->toChars() , global.hdr_ext);
+	}
 
     docfile = new File(docfilename);
 }
diff --git a/src/toobj.c b/src/toobj.c
index a8071e8..213e7a8 100644
--- a/src/toobj.c
+++ b/src/toobj.c
@@ -150,7 +150,7 @@ void Module::genmoduleinfo()
                  * they resolve to 0 if not pulled in by something else.
                  * Don't pull in a module just because it was imported.
                  */
-#if !OMFOBJ // Optlink crashes with weak symbols at EIP 41AFE7, 402000
+#if 1 || !OMFOBJ // Optlink crashes with weak symbols at EIP 41AFE7, 402000
                 s->Sflags |= SFLweak;
 #endif
                 dtxoff(&dt, s, 0, TYnptr);
diff --git a/src/win32.mak b/src/win32.mak
index 1a8fd19..8429b9d 100644
--- a/src/win32.mak
+++ b/src/win32.mak
@@ -8,7 +8,7 @@
 #   http://www.digitalmars.com/ctg/make.html
 # which should be in \dm\bin or in \dmd\windows\bin 
 
-D=
+D=z:
 DMDSVN=\svnproj\dmd\trunk\src
 #DMDSVN=\svnproj\dmd\branches\dmd-1.x\src
 SCROOT=$D\dm
@@ -51,9 +51,9 @@ defaulttarget: debdmd
 ################ RELEASES #########################
 
 release:
-	$(MAKE) clean
 	$(MAKE) dmd
-	$(MAKE) clean
+	copy dmd.exe Z:\dmd\bin\dmd2.exe
+	del *.exe *.map
 
 ################ NT COMMAND LINE RELEASE #########################
 
@@ -86,7 +86,7 @@ OBJ1= mars.obj enum.obj struct.obj dsymbol.obj import.obj id.obj \
 	glue.obj msc.obj ph.obj tk.obj s2ir.obj todt.obj e2ir.obj tocsym.obj \
 	util.obj eh.obj toobj.obj toctype.obj tocvdebug.obj toir.obj \
 	json.obj unittests.obj imphint.obj argtypes.obj apply.obj canthrow.obj \
-	sideeffect.obj
+	sideeffect.obj plugin.obj
 
 # from C/C++ compiler optimizer and back end
 
@@ -126,7 +126,7 @@ SRCS= mars.c enum.c struct.c dsymbol.c import.c idgen.c impcnvgen.c utf.h \
 	delegatize.c toir.h toir.c interpret.c traits.c builtin.c \
 	clone.c lib.h libomf.c libelf.c libmach.c arrayop.c intrange.c \
 	aliasthis.h aliasthis.c json.h json.c unittests.c imphint.c argtypes.c \
-	apply.c canthrow.c sideeffect.c
+	apply.c canthrow.c sideeffect.c plugin.c
 
 # From C++ compiler
 
